src/App.tsx

```diff
@@ -1,24 +1,291 @@
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todosStatus, setTodosStatus] = useState<Status>(Status.All);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<Message | ''>('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [titleTodo, setTitleTodo] = useState('');
+  const [deletedTodo, setDeletedTodo] = useState<number[] | null>(null);
+  const [changedTodo, setChangedTodo] = useState<number[] | null>(null);
+
+  const filterTodos = (listTodos: Todo[], status: Status) => {
+    switch (status) {
+      case Status.Active:
+        return listTodos.filter(todo => !todo.completed);
+      case Status.Completed:
+        return listTodos.filter(todo => todo.completed);
+      case Status.All:
+      default:
+        return listTodos;
+    }
+  };
+
+  const addTodo = ({ userId, title, completed }: Todo) => {
+    setErrorMessage('');
+    setIsLoading(true);
+
+    const createdTodo = {
+      id: 0,
+      userId: USER_ID,
+      title: title.trim(),
+      completed: false,
+    };
+
+    setTempTodo(createdTodo);
+
+    todoService.createTodo({ userId, title, completed })
+      .then(newTodo => {
+        setTodos(currentTodos => [...currentTodos, newTodo]);
+        setTitleTodo('');
+      })
+      .catch(() => {
+        setErrorMessage(Message.NoAddTodo);
+      })
+      .finally(() => {
+        setIsLoading(false);
+        setTempTodo(null);
+      });
+  };
+
+  const removeComplatedTodos = (todosId: number[] | null) => {
+    setIsLoading(true);
+
+    if (todosId) {
+      setDeletedTodo(todosId);
+      todosId.map(todoId => {
+        return todoService.deleteTodos(todoId)
+          .then(() => {
+            setTodos(currentTodos => (
+              currentTodos.filter(todo => todo.id !== todoId)
+            ));
+          })
+          .catch((error) => {
+            setDeletedTodo(null);
+            setErrorMessage(Message.NoDeleteTodo);
+            throw error;
+          })
+          .finally(() => {
+            setIsLoading(false);
+            setDeletedTodo(null);
+          });
+      });
+    }
+  };
+
+  const updateTodoStatus = (updatedTodo: Todo) => {
+    setErrorMessage('');
+    setIsLoading(true);
+    setChangedTodo([updatedTodo.id]);
+
+    return todoService.updateTodo(updatedTodo)
+      .then(() => {
+        setTodos(todos.map(todo => (
+          todo.id === updatedTodo.id
+            ? {
+              ...todo,
+              completed: updatedTodo.completed,
+
+            }
+            : todo
+        )));
+      })
+
+      .catch(() => setErrorMessage(Message.NoUpdateTodo))
+      .finally(() => {
+        setIsLoading(false);
+        setChangedTodo(null);
+      });
+  };
+
+  const updateTodosAllStatus = (updatedTodos: Todo[]) => {
+    setErrorMessage('');
+    setIsLoading(true);
+    const changedTodoId: number[] = [];
```

Comment:
pay attention to naming please
```suggestion
    const changedTodoIds: number[] = [];
```